doc/neps/nep-0023-backwards-compatibility.rst

```diff
@@ -19,46 +19,222 @@ processes for individual cases where breaking backwards compatibility
-- Backwards incompatible changes can be made, provided the benefits outweigh
-  the costs.
-- When assessing the costs, keep in mind that most users do not read the mailing
-  list, do not look at deprecation warnings, and sometimes wait more than one or
-  two years before upgrading from their old version.  And that NumPy has
-  many hundreds of thousands or even a couple of million users, so "no one will
-  do or use this" is very likely incorrect.
-- Benefits include improved functionality, usability and performance (in order
-  of importance), as well as lower maintenance cost and improved future
-  extensibility.
-- Bug fixes are exempt from the backwards compatibility policy.  However in case
-  of serious impact on users (e.g. a downstream library doesn't build anymore),
-  even bug fixes may have to be delayed for one or more releases.
-- The Python API and the C API will be treated in the same way.
-
-
-Examples
-^^^^^^^^
-
-We now discuss a number of concrete examples to illustrate typical issues
-and trade-offs.
+In addition, this NEP can serve as documentation for users about how the NumPy
+project treats backwards compatibility, and the speed at which they can expect
+changes to be made.
+
+In scope for this NEP are:
+
+- Principles of NumPy's approach to backwards compatibility.
+- How to deprecate functionality, and when to remove already deprecated
+  functionality.
+- Decision making process for deprecations and removals.
+- How to ensure that users are well informed about any change.
+
+Out of scope are:
+
+- Making concrete decisions about deprecations of particular functionality.
+- NumPy's versioning scheme.
+
+
+General principles
+------------------
+
+When considering proposed changes that are backwards incompatible, the
+main principles the NumPy developers use when making a decision are:
+
+1. Changes need to benefit more than they harm users.
+2. NumPy is widely used so breaking changes should by default be assumed to be
+   fairly harmful.
+3. Decisions should be based on how they affect users and downstream packages.
+   This should be based on usage data where possible. It does not matter whether
+   this use contradicts the documentation or best practices.
+4. The possibility of an incorrect result is much worse than an error or even crash.
+
+When assessing the costs of proposed changes, keep in mind that most users do
+not read the mailing list, do not notice deprecation warnings, and sometimes
+wait more than one or two years before upgrading from their old version. And
+that NumPy has millions of users, so "no one will do or use this" is very
+likely incorrect.
+
+Benefits of proposed changes can include improved functionality, usability and
+performance, as well as lower maintenance cost and improved future
+extensibility.
+
+Fixes for clear bugs are exempt from this backwards compatibility policy.
+However in case of serious impact on users (e.g. a downstream library doesn't
```

Comment:
Comma after "However". Maybe split into two sentences

"However, in case of serious impact on users even bug fixes may have to be delayed for one or more releases. For example, if a downstream library would no longer build or would give incorrect results."